contracts/governance/src/collection.rs

```diff
@@ -5,39 +5,117 @@
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
 use alloc::vec::Vec;
-use canonical::CanonError;
 use canonical_derive::Canon;
 
 #[derive(Clone, Canon, Debug)]
-pub struct Collection<K, V> {
+pub struct Map<K, V> {
     data: Vec<(K, V)>,
 }
 
-impl<K: PartialEq, V: PartialEq> Collection<K, V> {
-    // Methods return a result to keep things consistent with the map methods
-    pub fn get(&self, key: &K) -> Result<Option<&V>, CanonError> {
-        Ok(self.data.iter().find_map(|(k, v)| (k == key).then_some(v)))
+#[derive(Clone, Canon, Debug)]
+pub struct Set<V> {
+    data: Vec<V>,
+}
+
+impl<K: PartialEq, V: PartialEq> Map<K, V> {
+    #[allow(dead_code)]
+    pub fn get(&self, key: &K) -> Option<&V> {
+        self.data.iter().find_map(|(k, v)| (k == key).then_some(v))
+    }
+
+    pub fn get_mut(&mut self, key: &K) -> Option<&mut V> {
+        self.data
+            .iter_mut()
+            .find_map(|(k, v)| (k == key).then_some(v))
     }
 
-    pub fn insert(&mut self, key: K, value: V) -> Result<(), CanonError> {
+    pub fn insert(&mut self, key: K, value: V) {
         if let Some(pos) = self.data.iter().position(|(k, _)| k == &key) {
             self.data[pos] = (key, value)
         } else {
             self.data.push((key, value))
         }
-
-        Ok(())
     }
 
-    pub fn remove(&mut self, key: &K) -> Result<(), CanonError> {
+    #[allow(dead_code)]
+    pub fn remove(&mut self, key: &K) {
         self.data.retain(|(k, _)| k != key);
+    }
+}
+
+impl<V: PartialEq> Set<V> {
+    pub fn get(&self, value: &V) -> Option<&V> {
+        self.data.iter().find(|&v| v == value)
+    }
 
-        Ok(())
+    pub fn contains(&self, value: &V) -> bool {
+        self.data.iter().any(|v| v == value)
+    }
+
+    pub fn insert(&mut self, value: V) {
```

Comment:
nit: Should this return a bool indicating if the value has been inserted? (Like HashSet implementation does)